### PR TITLE
Merges develop into release (1.14.0 tag prep)

### PR DESCRIPTION
**Pull request by kbwestfall**
As titled.

Please do not merge until I've posted the results of the dev-suite tests.

### CODE REVIEW

**Review by profxj — APPROVED** [review]
249 files changes! :)

File: doc/dev/development.rst [hunk 1]
```diff
@@ -404,36 +404,45 @@ tagging process is as follows:
...
-
+ * A branch is created off of ``develop`` (typically called ``staged``) and then
+   a `PR <https://github.com/pypeit/PypeIt/compare>`_ is issued to merge
+   ``staged`` into ``release``.  This ``release...staged`` PR must meet the same
+   `Pull Request Acceptance Requirements`_ when merging new branches into
+   ``develop``.  Code review is expected to be limited (because all code changes
+   will have been reviewed before pulling into ``develop``), but the result of
+   the dev-suite tests must be shown and approved.  The reason for creating the
+   new branch instead of a direct ``release...develop`` PR is to allow for the
+   following updates to ``develop`` before merging (``develop`` is a protected
```
Comment (tbowers7) [inline doc/dev/development.rst]: Did you mean to say:
```
following updates to ``staged`` before merging (``develop`` is a protected
```
?

Comment (kbwestfall) [reply doc/dev/development.rst]: Yeah, good catch.  I see how that's confusing.  Changed.

**Comment by kbwestfall** [comment]
> 249 files changes! :)

Yeah, this includes the omnibus.

**Comment by kbwestfall** [comment]
First pass at tests were (finally) successful:
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  234 passed, 73 warnings in 399.53s (0:06:39) ---
--- PYTEST UNIT TESTS PASSED  123 passed, 193 warnings in 959.71s (0:15:59) ---
--- PYTEST VET TESTS PASSED  53 passed, 54 warnings in 967.68s (0:16:07) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 217/217 TESTS  ---
Testing Started at 2023-09-11T16:53:22.635773
Testing Completed at 2023-09-12T03:10:42.700263
Total Time: 10:17:20.064490
```
Still to do:
 - Merge #1664 into `develop`
 - Merge `develop` into `staged`
 - Rerun dev-suite

File: setup.cfg [hunk 2]
```diff
@@ -32,17 +32,17 @@ python_requires = >=3.9,<3.12
...
-    matplotlib>=3.3
+    matplotlib>=3.7
     PyYAML>=5.1
     configobj>=5.0.6
     scikit-learn>=1.0
     IPython>=7.10.0
-    ginga>=3.2
+    ginga>=4.1.1
     linetools
     qtpy>=1.9
```
Comment (tbowers7) [inline setup.cfg]: Don't know if it matters, but `ginga` requires `qtpy>=2.0.1` (released Feb 2022) for compatibility with PyQt6.

Comment (kbwestfall) [reply setup.cfg]: Thanks for catching this!  I'll bump this, too.

**Comment by kbwestfall** [comment]
Tests pass!

```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  239 passed, 73 warnings in 397.26s (0:06:37) ---
--- PYTEST UNIT TESTS PASSED  123 passed, 193 warnings in 973.80s (0:16:13) ---
--- PYTEST VET TESTS PASSED  53 passed, 68 warnings in 945.05s (0:15:45) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 217/217 TESTS  ---
Testing Started at 2023-09-12T21:45:12.552434
Testing Completed at 2023-09-13T07:44:47.045941
Total Time: 9:59:34.493507
```

@profxj , You'll need to override the blocked merge.  The tests that are hung are because of the changes to CI that Tim put in a while ago.

**Review by tbowers7 — APPROVED** [review]
Looks good!

**Comment by kbwestfall** [comment]
There are some test failures to work through:
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  239 passed, 73 warnings in 641.35s (0:10:41) ---
--- PYTEST UNIT TESTS PASSED  133 passed, 172 warnings in 1195.56s (0:19:55) ---
--- PYTEST VET TESTS FAILED  3 failed, 55 passed, 34 warnings in 1124.76s (0:18:44) ---
--- PYPEIT DEVELOPMENT SUITE FAILED 4/231 TESTS  ---
Failed tests:
    keck_lris_red/multi_1200_9000_d680_1x2 pypeit
    keck_lris_red/multi_1200_9000_d680 pypeit
    vlt_xshooter/VIS_1x1_LTT3218 pypeit_sensfunc
    vlt_xshooter/VIS_1x1_Feige110 pypeit_sensfunc
Skipped tests:
Testing Started at 2023-09-14T01:32:38.650865
Testing Completed at 2023-09-14T22:33:29.081505
Total Time: 21:00:50.430640
```

I think the vet test failures are a consequence of the other failures.

**Comment by kbwestfall** [comment]
Errors for Keck/LRISr were because of a filename change, and the VLT/XShooter errors were because of a download timeout.  I've pushed the relevant code fixes and restarted the dev-suite.

**Comment by kbwestfall** [comment]
Tests pass!

```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  239 passed, 70 warnings in 716.40s (0:11:56) ---
--- PYTEST UNIT TESTS PASSED  133 passed, 171 warnings in 1373.39s (0:22:53) ---
--- PYTEST VET TESTS PASSED  58 passed, 69 warnings in 1333.03s (0:22:13) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 231/231 TESTS  ---
Testing Started at 2023-09-15T18:49:04.245812
Testing Completed at 2023-09-17T04:46:04.732872
Total Time: 1 day, 9:57:00.487060
```

**Pull request merged**